factory/alchemy.py

```diff
@@ -48,7 +102,10 @@ def _create(cls, model_class, *args, **kwargs):
         obj = model_class(*args, **kwargs)
         if session is None:
             raise RuntimeError("No session provided.")
-        session.add(obj)
+        if cls._meta.sqlalchemy_get_or_create:
+            obj = cls._get_or_create(model_class, *args, **kwargs)
+        if obj not in session:
```

Comment:
Perhaps adding the object to the session should be done within `_get_or_create`. I’m not familiar with SQLAlchemy, but what about the following rule: when the object is created, add it to the session. When the object is gotten from the DB, check if it is in the session and adding if it wasn’t.

Reply:
Thank you for your input!

Session is only needed when adding or updating objects before persisting them to the database. (~commit)

Anyways, using your suggested `else:` condition made the code a little bit cleaner removing the `if obj in session:` check, it was only needed to avoid adding the object twice.

Let me know what you think.

Reply:
This is looking good to me, thanks for updating!